test/suite/test_bug031.py

```diff
@@ -0,0 +1,146 @@
+        cursor.remove()
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(20))
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # After eviction, we have:
+        # - Nothing in memory.
+        # - A record with a start ts 10 and stop ts 20 in the DS.
+        # - Nothing in the HS.
+
+        # Insert an update.
+        self.session.begin_transaction()
+        cursor[key] = value
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(30))
+
+        # Checkpoint.
+        self.session.checkpoint()
+
+        # After the checkpoint, we expect the following:
+        # - In memory, an update list with insert @ 30 (U3), remove @ 20 (U2) and insert @ 10 (U1).
+        # The two updates U1 and U2 should have the flag WT_UPDATE_PREPARE_RESTORED_FROM_DS as they
+        # were read back into memory from disk by the checkpoint.
+        # - The most recent insertion U3 in the DS.
+        # - A record with a start ts 10 and stop ts 20 in the HS.
+
+        # RTS.
+        self.conn.rollback_to_stable()
+
+        # Since the stable timestamp is 10, RTS marks all the updates performed at a time greater
+        # time than 10 as aborted. The update chain in memory is:
+        # U3 (aborted) -> U2 (aborted) -> U1.
+        # The DS now contains the insert @ 10 (U1) and the HS is empty.
+
+        # Insert but don't commit yet.
+        self.session.begin_transaction()
+        cursor[key] = value
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # Since only the insert @ 10 (U1) is (re)written to disk, we now only have the two aborted
+        # updates in memory: U3 (aborted) -> U2 (aborted). The rest remains unchanged.
+
+        # Now commit a new insertion.
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(40))
+
+        # We now have an additional insert @ 40 (U4) in the update list in memory on top of the two
+        # aborted updates: U4 -> U3 (aborted) -> U2 (aborted)
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.
+    
+        # Read update at timestamp 10.
+        # The update chain is U4 -> U3 (aborted) -> U2 (aborted). Without the fix in WT-10522, we
+        # would exit early because of the flag WT_UPDATE_RESTORED_FROM_DS set on U2 and miss on
```

Comment:
This sentence here is the entire crux of the bug. It isn't clearly called out. Please update with more details like:
Without the fix in WT-10522, the search call would return early with the XXX error instead of success because
the WT_UPDATE_RESTORED_FROM_DS flag is set on U2 and ...

Reply:
Added more details in 183179f07305e57a80978edf887c5a1e28243a40. Let me know what you think.

Reply:
Better explanation in 83634e682ce100ce7e77ea328c5cc9de72ff3eee